code/datums/wounds/burns.dm

```diff
@@ -0,0 +1,334 @@
+	/// Our current counter for how much flesh regeneration we have stacked from regenerative mesh/synthflesh/whatever, decrements each tick and lowers flesh_damage
+	var/flesh_healing = 0
+	/// How much dead flesh there is to scrape off before we can really start treatment to regenerate flesh, only relevant for severe and critical burns since moderate is only 2nd degree
+	var/mortification = 0
+
+		// Infestation vars (only for severe and critical)
+	/// How quickly infection breeds on this burn if we don't have disinfectant
+	var/infestation_rate = 0
+	/// Our current level of infection
+	var/infestation = 0
+	/// Our current level of sanitization/anti-infection, from disinfectants/alcohol/UV lights. While positive, totally pauses and slowly reverses infestation effects each tick
+	var/sanitization = 0
+
+	/// Once we reach infestation beyond WOUND_INFESTATION_SEPSIS, we get this many warnings before the limb is completely paralyzed (you'd have to ignore a really bad burn for a really long time for this to happen)
+	var/strikes_to_lose_limb = 3
+
+	/// The current bandage we have for this wound (maybe move bandages to the limb?)
+	var/obj/item/stack/current_bandage
+
+// TODO: flesh out (haha flesh), also clean up all of this to be more modular and less sprawly
+/datum/wound/burn/handle_process()
+	. = ..()
+	if(victim.reagents)
+		if(victim.reagents.has_reagent(/datum/reagent/medicine/spaceacillin))
+			sanitization += 0.4
+		if(victim.reagents.has_reagent(/datum/reagent/space_cleaner/sterilizine/))
+			sanitization += 0.4
+
+
+	if(current_bandage)
+		current_bandage.absorption_capacity -= WOUND_BURN_SANITIZATION_RATE
+		if(current_bandage.absorption_capacity <= 0)
+			victim.visible_message("<span class='danger'>Pus soaks through \the [current_bandage] on [victim]'s [limb.name].</span>", "<span class='warning'>Pus soaks through \the [current_bandage] on your [limb.name].</span>", vision_distance=COMBAT_MESSAGE_RANGE)
+			QDEL_NULL(current_bandage)
+			treat_priority = TRUE // todo: check if burns need this really
+
+	if(flesh_healing > 0)
+		var/bandage_factor = (current_bandage ? current_bandage.splint_factor : 1)
+		flesh_damage = max(0, flesh_damage - min(1, flesh_healing))
+		flesh_healing = max(0, flesh_healing - bandage_factor) // good bandages multiply the length of flesh healing
+
+	if((flesh_damage <= 0) && (infestation <= 0) && (mortification <= 0))
+		to_chat(victim, "<span class='green'>The burns on your [limb.name] have cleared up!</span>")
+		qdel(src)
+		return
+
+	if(sanitization > 0)
+		var/bandage_factor = (current_bandage ? current_bandage.splint_factor : 1)
+		infestation = max(0, infestation - WOUND_BURN_SANITIZATION_RATE)
+		sanitization = max(0, sanitization - (WOUND_BURN_SANITIZATION_RATE * bandage_factor))
+		return
+
+	infestation += infestation_rate
+
+	// TODO: actual math on this stuff
+	switch(infestation)
+		if(0 to WOUND_INFECTION_MODERATE)
+			if(mortification <= 0 && current_bandage)
+				flesh_healing += 0.1
```

Comment:
shouldn't this be BEFORE the return that having a positive sanitization value causes?